test/problems/poisson_regression.jl

```diff
@@ -0,0 +1,319 @@
+            #$@test g ≈ OptimizationMethods.grad(progData, x)
+            #@test progData.counters.neval_grad == nevals_grad 
+            #nevals_grad += 1
+
+            #TODO: Produces Error
+            #@test g ≈ OptimizationMethods.grad(progData, precomp, x)
+            #@test progData.counters.neval_grad == nevals_grad 
+            #nevals_grad += 1
+
+            OptimizationMethods.grad!(progData, precomp, store, x)
+            #TODO: Produces Error 
+            #@test g ≈ store.grad
+            @test progData.counters.neval_grad == nevals_grad 
+            nevals_grad += 1
+        end
+
+        ####################################
+        # Test Methods: Objective-Gradient Evaluation 
+        ####################################
+        for x in arg_tests
+            obj = objective(x)
+            gra = ForwardDiff.gradient(objective, x)
+            
+            # Without Precomputation 
+            o, g = OptimizationMethods.objgrad(progData, x)
+            @test o ≈ obj
+            #TODO: Produces Error 
+            #@test g ≈ gra
+            @test progData.counters.neval_obj == nevals_obj 
+            @test progData.counters.neval_grad == nevals_grad 
+            nevals_obj += 1
+            nevals_grad += 1
+
+
+            # With Precomputation 
+            o, g = OptimizationMethods.objgrad(progData, precomp, x)
+            @test o ≈ obj 
+            #TODO: Produces error 
+            #@test g ≈ gra
+            @test progData.counters.neval_obj == nevals_obj 
+            @test progData.counters.neval_grad == nevals_grad 
+            nevals_obj += 1
+            nevals_grad += 1
+
+
+            # With Precomputation and Allocation 
+            o = OptimizationMethods.objgrad!(progData, precomp, store, x)
+            @test o ≈ obj
+            #TODO: Produces Error 
+            #@test store.grad ≈ gra
+            @test progData.counters.neval_obj == nevals_obj 
+            @test progData.counters.neval_grad == nevals_grad 
+            nevals_obj += 1
+            nevals_grad += 1
+
+        end
+
+        ####################################
+        # Test Methods: Hessian 
+        ####################################
+        for x in arg_tests
+            h = ForwardDiff.hessian(objective, x)
+            
+            # Without Precomputation 
+            @test h ≈ OptimizationMethods.hess(progData, x)
+            @test progData.counters.neval_hess == nevals_hess 
+            nevals_hess += 1
+
+            # With Precomputation 
+            @test h ≈ OptimizationMethods.hess(progData, precomp, x)
+            @test progData.counters.neval_hess == nevals_hess 
+            nevals_hess += 1
+
+            # With Precomputation and Allocation 
+            #TODO: Produces Errors 
+            #OptimizationMethods.hess!(progData, precomp, store, x)
+            #@test h ≈ store.hess
+            #@test progData.counters.neval_hess == nevals_hess 
+            #nevals_hess += 1
```

Comment:
How come these are producing errors

Reply:
Oh I see, this is my bug.

Reply:
`OptimizationMethods.hess!` was not filling the stored hessian with zeros, so it was accumulating.